antarest/study/storage/variantstudy/model/command/generate_thermal_cluster_timeseries.py

```diff
@@ -0,0 +1,149 @@
+        general_data = study_data.tree.get(["settings", "generaldata"], depth=3)
+        thermal_seed = general_data["seeds - Mersenne Twister"]["seed-tsgen-thermal"]
+        nb_years = general_data["general"]["nbtimeseriesthermal"]
+        # 2 - Build the generator
+        rng = MersenneTwisterRNG(seed=thermal_seed)
+        generator = ThermalDataGenerator(rng=rng, days=365)
+        # 3- Loop through areas in alphabetical order
+        areas: t.Dict[str, Area] = study_data.config.areas
+        sorted_areas = {k: areas[k] for k in sorted(areas)}
+        for area_id, area in sorted_areas.items():
+            # 4- Loop through thermal clusters in alphabetical order
+            sorted_thermals = sorted(area.thermals, key=lambda x: x.id)
+            for thermal in sorted_thermals:
+                # 5 - Filters out clusters with no generation
+                if thermal.gen_ts == LocalTSGenerationBehavior.FORCE_NO_GENERATION:
+                    continue
+                # 6- Build the cluster
+                modulation_matrix = study_data.tree.get(
+                    ["input", "thermal", "prepro", area_id, thermal.id.lower(), "modulation"]
+                )["data"]
+                modulation_capacity = np.array([row[2] for row in modulation_matrix])
+                ts_generator_matrix = study_data.tree.get(
+                    ["input", "thermal", "prepro", area_id, thermal.id.lower(), "data"]
+                )["data"]
+                fo_duration, po_duration, fo_rate, po_rate, npo_min, npo_max = [
+                    np.array(col, dtype=int) if i not in [2, 3] else np.array(col, dtype=float)
+                    for i, col in enumerate(zip(*ts_generator_matrix))
+                ]
+                cluster = ThermalCluster(
+                    unit_count=thermal.unit_count,
+                    nominal_power=thermal.nominal_capacity,
+                    modulation=modulation_capacity,
+                    fo_law=ProbabilityLaw(thermal.law_forced.value.upper()),
+                    fo_volatility=thermal.volatility_forced,
+                    po_law=ProbabilityLaw(thermal.law_planned.value.upper()),
+                    po_volatility=thermal.volatility_planned,
+                    fo_duration=fo_duration,
+                    fo_rate=fo_rate,
+                    po_duration=po_duration,
+                    po_rate=po_rate,
+                    npo_min=npo_min,
+                    npo_max=npo_max,
+                )
+                # 7- Generate the time-series
+                results = generator.generate_time_series(cluster, nb_years)
+                generated_matrix = results.available_power.T.tolist()
+                # 8- Generates the UUID for the `get_inner_matrices` method
+                uuid = study_data.tree.context.matrix.create(generated_matrix)
+                self._INNER_MATRICES.append(uuid)
+                # 9- Write the matrix inside the input folder.
```

Comment:
It's a shame that we need to:
1. write the matrix twice, in the matrix store AND as a file here
2. duplicate here the code that writes a matrix to a file: it's only 1 line but we need to ensure consistency between here and other places where we write matrice, it's error-prone for future evolutions

For 2, we should probably have a small utility method shared at least beween `matrix.py` and here.
For 1, I have not easy solution, we probaby need to postpone this ... Note that this kind of direct write will be an issue the day we want to store input matrices with a different format.

Reply:
Now that we don't have inner_matrices anymore, the 1. is not relevant.